### PR TITLE
RC 3.12.0

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

Release candidate for 3.12.0

## What type(s) of changes does this code introduce?

- Bugfix
- New feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- Build related changes
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes

### Does this introduce a breaking change?

- Yes. Minimum versions are increased.

## What systems has this change been tested on?

None. Nightlies were OK last night and no "significant" change is believed to have been merged recently.

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
I built and verified this PR independently. Given the python breakage on sulfur the test failures all look consistent with the nightlies.  Other than the ongoing python-NEXUS troubles, none of the deterministic tests failed.

I also verified that the version is correctly reported as v3.12.0

```
cmake -GNinja -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DQMC_DATA=/scratch/pk7/QMC_DATA_FULL/ -DQMC_NIO_MAX_SIZE=16 ../qmcpack/
ninja
nohup ctest -j 48 -E long --output-on-failure >& ctest.out &
```

```
96% tests passed, 72 tests failed out of 1845

Label Time Summary:
QMCPACK                     = 58610.30 sec*proc (201 tests)
QMCPACK-checking-results    = 116.75 sec*proc (1315 tests)
abort                       = 2822.68 sec*proc (24 tests)
bug                         = 8263.15 sec*proc (58 tests)
cause_unknown               = 14807.38 sec*proc (131 tests)
check_fail                  = 2249.10 sec*proc (19 tests)
converter                   =  56.87 sec*proc (24 tests)
coverage                    = 116.35 sec*proc (5 tests)
definite_stat_fail          = 1671.87 sec*proc (10 tests)
deterministic               = 570.42 sec*proc (1029 tests)
hard_fail                   = 1530.62 sec*proc (11 tests)
intermittent_stat_fail      = 14807.38 sec*proc (131 tests)
nexus                       =  77.73 sec*proc (69 tests)
performance                 = 4954.86 sec*proc (78 tests)
poor_test                   =  24.37 sec*proc (14 tests)
quality_unknown             = 65025.93 sec*proc (1722 tests)
unit                        =  29.18 sec*proc (35 tests)
unstable                    =  54.51 sec*proc (516 tests)

Total Test time (real) = 1491.45 sec

The following tests FAILED:
         66 - estimator-latdev-1-16-check (Failed)
         70 - estimator-sofk_pbyp-properties-1-16-check (Failed)
         72 - estimator-sofk_pbyp-collectables-1-16-check (Failed)
         74 - estimator-sofk_allp-properties-1-16-check (Failed)
         76 - estimator-sofk_allp-collectables-1-16-check (Failed)
        283 - short-C4_md-ae-excited-qp-vmc-1-16-variance (Failed)
        290 - short-C4_md-ae-ground-qp-vmc-1-16-variance (Failed)
        300 - short-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4-check (Failed)
        394 - short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4 (Failed)
        395 - short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4-check (Failed)
        397 - short-LiH_pp-vmc_hf_sdj_estimator_energydensity_voronoi-4-4-check (Failed)
        413 - short-H4-RHF-orb-opt-16-1-totenergy (Failed)
        417 - short-H4-orb-opt-1-16-totenergy (Failed)
        418 - short-H4-orb-opt-dmc-16-1 (Failed)
        419 - short-H4-orb-opt-dmc-16-1-totenergy (Failed)
        421 - short-H4-orb-opt-dmc-1-16-totenergy (Failed)
        445 - short-H4-msd-orb-opt-16-1-totenergy (Failed)
        469 - short-chn_ae_cuspCorrection-vmc-1-16-kinetic (Failed)
        471 - short-chn_ae_cuspCorrection-vmc-1-16-eeenergy (Failed)
        632 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1 (Failed)
        633 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-totenergy_A (Failed)
        634 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-totenergy_B (Failed)
        635 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-dtotenergy_AB (Failed)
        636 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-ionion_A (Failed)
        637 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-ionion_B (Failed)
        638 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-dionion_AB (Failed)
        639 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-eeenergy_A (Failed)
        640 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-eeenergy_B (Failed)
        641 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-deeenergy_AB (Failed)
        642 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1 (Failed)
        643 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-totenergy_A (Failed)
        644 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-totenergy_B (Failed)
        645 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-dtotenergy_AB (Failed)
        646 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-ionion_A (Failed)
        647 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-ionion_B (Failed)
        648 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-dionion_AB (Failed)
        649 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-eeenergy_A (Failed)
        650 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-eeenergy_B (Failed)
        651 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-deeenergy_AB (Failed)
        652 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1 (Failed)
        653 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-totenergy_A (Failed)
        654 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-totenergy_B (Failed)
        655 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-dtotenergy_AB (Failed)
        656 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-ionion_A (Failed)
        657 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-ionion_B (Failed)
        658 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-dionion_AB (Failed)
        659 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-eeenergy_A (Failed)
        660 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-eeenergy_B (Failed)
        661 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-deeenergy_AB (Failed)
        741 - short-diamondC_1x1x1_pp-vmc-dmc-allp_sdj-1-16-eeenergy (Failed)
        747 - short-diamondC_1x1x1_pp-vmc-dmc-allp_sdj-1-16-mpc (Failed)
        776 - short-diamondC_1x1x1_pp-vmcbatch-estimator-onebodydensitymatrices-4-4-check (Failed)
        780 - short-diamondC_1x1x1_pp-vmc-noJ-estimator-1rdm-4-4-check (Failed)
        782 - short-diamondC_1x1x1_pp-vmc-J2-estimator-1rdm-4-4-check (Failed)
        784 - short-diamondC_1x1x1_pp-vmc-estimator-energydensity-cell-4-4-check (Failed)
        786 - short-diamondC_1x1x1_pp-vmc-estimator-energydensity-voronoi-4-4-check (Failed)
        788 - short-diamondC_1x1x1_pp-dmc-estimator-energydensity-cell-4-4-check (Failed)
        1112 - short-diamondC_2x1x1_pp-delayed_update-vmc_sdj-1-16-totenergy (Failed)
        1129 - short-diamondC_2x1x1_pp-dmc_sdj-1-16-totenergy (Failed)
        1137 - short-diamondC_2x1x1_pp-vmcbatch_dmcbatch_4r_sdj-4-4-totenergy (Failed)
        1834 - ntest_nexus_nxs_sim (Failed)
        1835 - ntest_nexus_qmca (Failed)
        1836 - ntest_nexus_qmc_fit (Failed)
        1837 - ntest_nexus_qdens (Failed)
        1838 - ntest_nexus_example_gamess_H2O (Failed)
        1839 - ntest_nexus_example_pwscf_relax_Ge_T (Failed)
        1840 - ntest_nexus_example_qmcpack_H2O (Failed)
        1841 - ntest_nexus_example_qmcpack_LiH (Failed)
        1842 - ntest_nexus_example_qmcpack_c20 (Failed)
        1843 - ntest_nexus_example_qmcpack_diamond (Failed)
        1844 - ntest_nexus_example_qmcpack_graphene (Failed)
        1845 - ntest_nexus_example_qmcpack_oxygen_dimer (Failed)
Errors while running CTest
```

**Comment by prckent** [comment]
Since the release is dated tomorrow 8 December we can wait on merging (after approval) to allow for wider comment.

**Comment by prckent** [comment]
Start testing in-house

**Review by markdewing — APPROVED** [review]

**Pull request merged**
